src/components/Form/formFunctions.ts

```diff
@@ -0,0 +1,281 @@
+import React from "react";
+import { SelectChangeEvent } from "@mui/material";
+import { booleanGroup } from "@/components/DataInput/inputHandlerFactories";
+import supabase, { InsertSchema } from "@/supabase";
+import { Database } from "@/database_types_file";
+
+type Field = Fields[keyof Fields];
+type OnChange = (event: React.ChangeEvent<HTMLInputElement> | SelectChangeEvent) => void;
+type OnChangeCheckbox = (event: React.ChangeEvent<HTMLInputElement>) => void;
+export type ErrorSetter = (errorKey: string, errorType: Error) => void;
+export type FieldSetter = (fieldKey: string, newFieldValue: Field) => void;
+type PersonType = Database["public"]["Enums"]["gender"];
+type FamilyDatabaseRecord = InsertSchema["families"];
+export type ClientDatabaseRecord = InsertSchema["clients"];
+export enum Error {
```

Comment:
What is the benefit of using enum here as opposed to a regular const?

Reply:
I think it makes more sense to treat the errors as "a group of constants belonging to the `Error` family" rather than "a constant (object/interface) with these properties". 

Other than that, there's not much difference with using enums and consts, except that an enum can be used as a type and the values that each member are initialised to are fixed and different from each other (which I think is suitable for use here as each error is different).